contract/controllers/main.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+from odoo.addons.portal.controllers.portal import CustomerPortal,\
+    pager as portal_pager
+
+
+class PortalContract(CustomerPortal):
+
+    def _prepare_portal_layout_values(self):
+        values = super()._prepare_portal_layout_values()
+        model = 'contract.contract'
+        if request.env[model].check_access_rights('read', raise_exception=False):
+            values['contract_count'] = request.env[model].search_count([])
+        else:
+            values['contract_count'] = 0
+        return values
+
+    def _contract_get_page_view_values(self, contract, access_token, **kwargs):
+        values = {
+            'page_name': 'Contracts',
```

Comment:
Shouldn't this go translated?
```suggestion
            "page_name": _("Contracts"),
```
I also advise you to get used to double quotes, or cherry-picking this to v13 will mean a lot of changes.